src/deepsparse/v2/pipeline.py

```diff
@@ -13,88 +13,113 @@
+        context to reflect each step of the router. The input to a given operator is the
+        output of the previous operator.
+
+        :param inp: input to the operator. expected to be of any type that is
+        expected by the operator.
+        :param context: context to store the current the inputs, outputs, and operator
+        for each step of the router.
+
+        """
+        next_step = self.router.START_ROUTE
+        while next_step != self.router.END_ROUTE:
+            # Either a dictionary key or valid index
+            operator = self.ops[next_step]
+
+            output_future = self._scheduler_group.submit(
+                operator=operator, operator_input=inp, context=context
+            )
+
+            # wait for future to resolve
+            operator_output = output_future.result()
+
+            # update context
+            context.update(
+                operator=operator,
+                input=inp,
+                output=operator_output,
+            )
+
+            next_step = self.router.next(next_step, self.ops)
+            inp = operator_output
+        return operator_output, context
+
     def __call__(self, *args, return_context: bool = False, **kwargs):
         """
-        :param return_context: if True, retrns tuple of the pipelien output
+        :param return_context: if True, returns tuple of the pipeline output
             and entire context. Default False
         :return: output of the pipeline stages ran with the router for the given input
         """
         if len(args) > 1:
             raise ValueError(
-                "Only 1 in-line argument may be supplied to Pipeline which "
-                f"must be a Schema, found: {len(args)}"
+                "Only 1 unnamed arg may be supplied to a Pipeline, the input expected"
+                "for the first Operator."
             )
         if args and kwargs:
             raise ValueError(
-                "Pipeline can only run either a single in-line argument schema or a "
+                "Pipeline can only run either a in-line arguments or a "
                 f"series of kwargs, found {len(args)} args and {len(kwargs)} kwargs"
             )
 
-        pipeline_input = args[0] or kwargs
-        pipeline_output, context = self.router.run(
-            inp=pipeline_input,
-            operators=self.stages,
-            scheduler=self._scheduler_group,
-        )
+        pipeline_input = kwargs or args[0]
+        pipeline_output, context = self.run(inp=pipeline_input, context=Context())
 
         if return_context:
             return pipeline_output, context
 
         return pipeline_output
 
     def validate(self):
-        router_validation = self.router.validate(self.stages)
+        """
+        Validate that compatability of the router and operators provided.
+        """
+        router_validation = self.router.validate(self.ops)
```

Comment:
I see that you are trying to get rid of the name `stages`, I like that. Could we do a sweep through `v2` directory and made sure that all naming of `stages` is deprecated in favour of `ops`?